boa/contracts/vyper/vyper_contract.py

```diff
@@ -805,7 +805,7 @@ def line_profile(self, computation=None):
         for child in computation.children:
             child_obj = self.env.lookup_contract(child.msg.code_address)
             # TODO: child obj is opaque contract that calls back into known contract
-            if child_obj is not None:
+            if child_obj is not None and isinstance(child_obj, VyperContract):
```

Comment:
Maybe hasattr here too?

Reply:
Vvm contract should have line profiling soon and this could work for solidity too

Reply:
i think the entire recursion is broken actually, there is double recursion because there is recursion in profiling.py and also in this file